src/tests/slot-composer-test.ts

```diff
@@ -73,168 +84,95 @@ recipe
-
-    assert.lengthOf(slotComposer.getAvailableContexts(), 1);
 
-    const verifyContext = (name: string, expected) => {
-      const context = slotComposer.findContextsByName(name)[0];
-      assert.exists(context);
-      assert.strictEqual(expected.sourceSlotName, context.sourceSlotConsumer ? context.sourceSlotConsumer.consumeConn.name : undefined);
-      assert.strictEqual(expected.hasContainer, Boolean(context.container));
-      assert.deepEqual(expected.consumeConnNames || [], context.slotConsumers.map(slot => slot.consumeConn.getQualifiedName()));
-    };
-    verifyContext('root', {hasContainer: true});
-
-    plan = plan.clone();
+    let {arc, slotComposer, observer, plan} = await initSlotComposer(manifestStr);
 
     // instantiate the recipe
+    plan = plan.clone();
     plan.normalize();
     assert.isTrue(plan.isResolved());
     assert.strictEqual(arc.pec.slotComposer, slotComposer);
+
+    observer.newExpectations()
+      .expectRenderSlot('A', 'root')
+      .expectRenderSlot('B', 'mySlot')
+      .expectRenderSlot('BB', 'mySlot')
+      .expectRenderSlot('C', 'otherSlot')
+      ;
+
     await arc.instantiate(plan);
-    assert.deepEqual(['A'], startRenderParticles);
-
-    // TODO(sjmiles): contexts deprecated
-    // assert.lengthOf(slotComposer.getAvailableContexts(), 3);
-    // verifyContext('root', {hasContainer: true, consumeConnNames: ['A::root']});
-    // verifyContext('mySlot', {hasContainer: false, sourceSlotName: 'root', consumeConnNames: ['B::mySlot', 'BB::mySlot']});
-    // verifyContext('otherSlot', {hasContainer: false, sourceSlotName: 'root', consumeConnNames: ['C::otherSlot']});
-
-    // render root slot
-    const particle = arc.activeRecipe.particles[0];
-    const rootSlot = slotComposer.getSlotConsumer(particle, 'root');
-    const mySlotId = slotComposer.findContextsByName('mySlot')[0].id;
-    rootSlot.getInnerContainer = (slotId) => slotId === mySlotId ? 'dummy-inner-container' : null;
-    startRenderParticles.length = 0;
-    await slotComposer.renderSlot(particle, 'root', {model: {'foo': 'bar'}});
-    assert.deepEqual(['B', 'BB'], startRenderParticles);
-
-    // TODO(sjmiles): render data no longer captured by slot objects
-    //assert.deepEqual({foo: 'bar'}, rootSlot.getRendering().model);
-
-    // TODO(sjmiles): contexts deprecated
-    // assert.lengthOf(slotComposer.getAvailableContexts(), 3);
-    // verifyContext('root', {hasContainer: true, consumeConnNames: ['A::root']});
-    // verifyContext('mySlot', {hasContainer: true, sourceSlotName: 'root', consumeConnNames: ['B::mySlot', 'BB::mySlot']});
-    // verifyContext('otherSlot', {hasContainer: false, sourceSlotName: 'root', consumeConnNames: ['C::otherSlot']});
-
-    // TODO(sjmiles): uses old render data, will be repaired in subsequent PR
-    //await slotComposer.expectationsCompleted();
+    await arc.idle;
+    await observer.expectationsCompleted();
   });
 
+  // TODO(sjmiles): missing info on why this is skipped
```

Comment:
fyi i'll look into this tomorrow and try to find out why this happened.

Reply:
After we talked, the notion is that some of these were skipped because they were failing at a critical time and it was easier to skip than to root-cause, so it's tech debt.